rust/frontend/src/module.rs

```diff
@@ -0,0 +1,110 @@
+//! Provides [`Module`] type and methods for working with runtime TVM modules.
+
+use std::ffi::CString;
+use std::mem;
+use std::os::raw::{c_char, c_int};
+use std::path::Path;
+use std::ptr;
+
+use ts;
+
+use function::{self, Function};
+use internal_api;
+use Error;
+use Result;
+
+const ENTRY_FUNC: &'static str = "__tvm_main__";
+
+/// Wrapper around TVM module handle.
+#[derive(Debug, Clone)]
+pub struct Module {
+    pub(crate) handle: ts::TVMModuleHandle,
+    is_released: bool,
+    pub(crate) entry: Option<Function>,
+}
+
+impl Module {
+    pub(crate) fn new(
+        handle: ts::TVMModuleHandle,
+        is_released: bool,
+        entry: Option<Function>,
+    ) -> Self {
+        Self {
+            handle,
+            is_released,
+            entry,
+        }
+    }
+
+    /// Sets the entry function of a module.
+    pub fn entry_func(mut self) -> Self {
```

Comment:
a function which takes `mut self` and returns `Self` is odd. `(&mut self) -> ()` is more transparent about what it's doing